codegen/builtin.go

```diff
@@ -0,0 +1,129 @@
+package codegen
+
+type Function interface{}
+
+var BuiltinFunctions = map[string]Function{
```

Comment:
Should these files be in the `builtin` package?

Reply:
Yeah I'm debating on moving the `builtin` package as well, so currently in flux. But as it stands, yes I should move this file over.